lib/screens/settings/stores/settings_store.dart

```diff
@@ -301,6 +310,15 @@ abstract class _SettingsStoreBase with Store {
   @observable
   var darkenRecentMessages = defaultDarkenRecentMessages;
 
+  @JsonKey(defaultValue: getDefaultMutedWords)
+  @ObservableMutedWordsListConverter()
+  @observable
+  ObservableList<String> mutedWords = getDefaultMutedWords();
```

Comment:
```suggestion
  @JsonKey(defaultValue: getDefaultMutedWords)
  @observable
  var mutedWords = defaultMutedWords;
```

From thereon we can rely on immutability instead of mutations for the list changes. So anytime we want to edit the list just reassign the whole thing and Mobx will detect the changes.